### PR TITLE
Fixed sessions CSV export

### LINKED ISSUE #5519: Session Export CSV does not include all data
The Session Export should export all data sets that are available e.g. including:
* Submission time
* All speakers
* Proposed length
* Type (Workshop, Talk)
* Level (e.g. Intermediate)
* Status (e.g. pending, accepted etc.)

screenshot from 2019-01-17 00-04-14

### PR DESCRIPTION
**Pull request by mrsaicharan1**
#### Checklist
Fixes #5519 

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
- Added documentation for test database

#### Changes proposed in this pull request:

- Documentation additions

- Added functionality for exporting CSV data as stream

### CODE REVIEW

File: app/api/helpers/csv_jobs_util.py [hunk 1]
```diff
@@ -47,9 +48,11 @@ def export_attendees_csv(attendees):
 
 def export_sessions_csv(sessions):
     headers = ['Session Title', 'Session Speakers',
-               'Session Track', 'Session Abstract', 'Created At', 'Email Sent']
+               'Session Track', 'Session Abstract', 'Created At', 'Email Sent',
+               'Level', 'Status', 'Session Type', 'Talk Length']
     rows = [headers]
     for session in sessions:
+        print(session)
```
Comment (srv-twry) [inline app/api/helpers/csv_jobs_util.py]: Use of this?

**Comment by srv-twry** [comment]
Are adding test database docs and fixing exports related? If not, kindly create a separate issue for database docs one and open a PR for that. Don't solve multiple unrelated issues in a single PR.

**Comment by mrsaicharan1** [comment]
@srv-twry , Yeah, will separate the PRs to match their relevant issues

**Comment by mrsaicharan1** [comment]
@CosmicCoder96 Could you follow up on this?

File: app/api/helpers/csv_jobs_util.py [hunk 2]
```diff
@@ -64,9 +66,14 @@ def export_sessions_csv(sessions):
             column.append(strip_tags(session.short_abstract) if session.short_abstract else '')
             column.append(session.created_at if session.created_at else '')
             column.append('Yes' if session.is_mail_sent else 'No')
+            column.append(session.level)
+            column.append(session.state)
+            column.append(session.type)
+            column.append(len(session.long_abstract))
             rows.append(column)
-
-    return rows
```
Comment (CosmicCoder96) [inline app/api/helpers/csv_jobs_util.py]: I maybe wrong, but only the missing fields needed to be appended, what does the `Response(stream_with_context` do?

Comment (mrsaicharan1) [reply app/api/helpers/csv_jobs_util.py]: There wasn't any csv which was being exported. So, I added that line to stream the csv for it.

**Comment by mrsaicharan1** [comment]
@

> No, check
> 
> [open-event-server/app/api/helpers/tasks.py](https://github.com/fossasia/open-event-server/blob/4c12e36efed66db4a308d30be40f3eba878e7198/app/api/helpers/tasks.py#L334)
> 
> Line 334 in [4c12e36](/fossasia/open-event-server/commit/4c12e36efed66db4a308d30be40f3eba878e7198)
>  def export_sessions_csv_task(self, event_id): 
> 
> The function you edited is only supposed to return the rows, not generate the CSV. Generating CSV is a time-consuming task, and cannot be allowed to run synchronously. Hence it is handled as an asynchronous celery task.
> 
> Upon carefully reading the issue, I can see that the issue wasn't an absence of generated CSV, but rather the presence of limited fields in it.

Oh okay! Didn't see that file there. Yea, so I'll remove the stream content and then send a PR!

**Comment by mrsaicharan1** [comment]
@CosmicCoder96 Is there anything else required?

**Comment by CosmicCoder96** [comment]
@mrsaicharan1 Yes, you have not made any changes since I last requested them.

**Comment by CosmicCoder96** [comment]
Also, squash your commits.

**Comment by mrsaicharan1** [comment]
@CosmicCoder96 I've squashed my commits for this PR.

File: app/api/helpers/csv_jobs_util.py (hunk 1 above)
Comment (mrsaicharan1) [reply app/api/helpers/csv_jobs_util.py]: Removed the unnecessary print statements.

**Comment by CosmicCoder96** [comment]
@mrsaicharan1 There are 30 commits.
image

**Pull request closed without merging**